lib/controllers/dialog-controller/index.js

```diff
@@ -67,24 +67,55 @@ module.exports = class DialogController extends SkillSimulationController {
      * @param {Function} callback
      */
     setupSpecialCommands(dialogReplView, callback) {
-        dialogReplView.registerRecordCommand((filePath) => {
-            if (!stringUtils.isNonBlankString(filePath)) {
-                return callback('A file name has not been specified');
+        dialogReplView.registerRecordCommand((recordArgs) => {
+            const recordArgsList = recordArgs.trim().split(' ');
+            const RECORD_FORMAT = 'Please use the format: ".record <fileName>" or ".record <fileName> --append-quit"';
+            if (!stringUtils.isNonBlankString(recordArgs) || recordArgsList.length > 2) {
+                return Messenger.getInstance().warn(`Incorrect format. ${RECORD_FORMAT}`);
             }
-            const JSON_FILE_EXTENSION = '.json';
-            if (path.extname(filePath).toLowerCase() !== JSON_FILE_EXTENSION) {
-                return callback(`File should be of type '${JSON_FILE_EXTENSION}'`);
+            const { filePath, shouldAppendQuit } = this._validateRecordCommandInput(recordArgsList, RECORD_FORMAT);
+            const utteranceCacheCopy = [...this.utteranceCache];
+            if (shouldAppendQuit) {
+                utteranceCacheCopy.push('.quit');
             }
-            try {
-                this.createReplayFile(filePath);
-                Messenger.getInstance().info(`Created replay file at ${filePath}`);
-            } catch (err) {
-                return callback(err);
+            if (filePath) {
+                try {
+                    this.createReplayFile(filePath, utteranceCacheCopy);
+                    Messenger.getInstance().info(`Created replay file at ${filePath}`
+                        + `${shouldAppendQuit ? ' (appended ".quit" to list of utterances).' : ''}`);
+                } catch (replayFileCreationError) {
+                    return callback(replayFileCreationError);
+                }
             }
         });
         dialogReplView.registerQuitCommand(() => {});
     }
 
+    /**
+     * Validate record command arguments.
+     * @param {Array} recordArgsList
+     * @param {String} recordCommandFormat
+     */
+    _validateRecordCommandInput(recordArgsList, recordCommandFormat) {
+        const filePath = recordArgsList[0];
+        const appendQuitArgument = recordArgsList[1];
+        let shouldAppendQuit = false;
+        const JSON_FILE_EXTENSION = '.json';
+
+        if (path.extname(filePath).toLowerCase() !== JSON_FILE_EXTENSION) {
+            Messenger.getInstance().warn(`File should be of type '${JSON_FILE_EXTENSION}'`);
+            return {};
+        }
+        if (stringUtils.isNonBlankString(appendQuitArgument)) {
```

Comment:
why do you need so much logic?
something like this would not be sufficient?

const appendQuitArgument = recordArgsList[1] === '--append-quit' || false;

Reply:
We'd want to show specific error messages for specific errors committed by developer. Hence the blocked `if-else` statement.